### PR TITLE
Replace PSVersion source generator with incremental one

### LINKED ISSUE #21228: Replace SourceGenerator with IIncrementalGenerator
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://learn.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

RS1035 warning is disabled in PSVersionSourceGenerator.cs. It uses SourceGenerator. It is deprecated and needs to be replaced with IIncrementalGenerator 

More info: https://github.com/dotnet/roslyn/blob/main/docs/features/incremental-generators.md

### Expected behavior

```console
IIncrementalGenerator needs to be implemented
```

### Actual behavior

```console
Remove SourceGenerator
```

### Error details

_No response_

### Environment data

```powershell
Latest source
```

### Visuals

_No response_

### PR DESCRIPTION
**Pull request by iSazonov**
# PR Summary

No functional changes. Only build fix #21228

## PR Checklist

- [ ] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [ ] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [ ] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/main/reference/7.3/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [ ] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [ ] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and opened an issue in the relevant tool repository. This may include:
        - [ ] Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode
        (which runs in a different PS Host).
            - [ ] Issue filed: 
        - [ ] Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
            - [ ] Issue filed: 
        - [ ] Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
            - [ ] Issue filed: 
        - [ ] Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).
            - [ ] Issue filed:

### CODE REVIEW

File: PowerShell.sln [hunk 1]
```diff
@@ -40,6 +40,8 @@ EndProject
 #
 # 73EA0BE6-C0C5-4B56-A5AA-DADA4C01D690 - powershell-unix
 # Only Linux is valid, all configurations mapped to Linux
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "PSVersionInfoGenerator", "src\System.Management.Automation\SourceGenerators\PSVersionInfoGenerator\PSVersionInfoGenerator.csproj", "{B22424E8-0516-4FC3-A9CB-D84D15EF0589}"
+EndProject
```
Comment (daxian-dbw) [inline PowerShell.sln:44]: Should be moved up with other `Project` declarations.

File: src/System.Management.Automation/SourceGenerators/PSVersionInfoGenerator/PSVersionInfoGenerator.cs [hunk 2]
```diff
@@ -11,40 +11,67 @@ namespace SMA
...
+                    MainVersion = mainVersion ?? string.Empty,
+                    GitDescribe = gitDescribe ?? string.Empty,
+                    ReleaseTag = releaseTag ?? string.Empty
+                };
+
+                return options;
+            });
+
+            context.RegisterSourceOutput(buildOptionsProvider,
+            static (context, buildOptions) =>
```
Comment (daxian-dbw) [inline src/System.Management.Automation/SourceGenerators/PSVersionInfoGenerator/PSVersionInfoGenerator.cs:42]: Minor: should have a level of indentation for this lambda function.

File: src/System.Management.Automation/SourceGenerators/PSVersionInfoGenerator/PSVersionInfoGenerator.cs [hunk 3]
```diff
@@ -11,40 +11,67 @@ namespace SMA
...
+                    CultureInfo.InvariantCulture,
+                    SourceTemplate,
+                    buildOptions.ProductVersion,
+                    gitCommitId,
+                    versions.major,
+                    versions.minor,
+                    versions.patch,
+                    versions.preReleaseLabel);
+
+                context.AddSource($"PSVersionInfo.g.cs", result);
```
Comment (daxian-dbw) [inline src/System.Management.Automation/SourceGenerators/PSVersionInfoGenerator/PSVersionInfoGenerator.cs:61]:
```suggestion
                context.AddSource("PSVersionInfo.g.cs", result);
```

**Review by daxian-dbw — APPROVED** [review]

**Comment by juarenesgrillonetto2014-beep** [comment]
"C:\Program Files\Google\Chrome\Application\chrome.exe" "c:\Users\fabio\Nova pasta\tao-bonita\index.html"

**Pull request merged**